starknet_py/tests/e2e/mock/compile_contracts.sh

```diff
@@ -1,32 +1,81 @@
 #!/bin/bash
-
+set -e
 MOCK_DIRECTORY=starknet_py/tests/e2e/mock
-CONTRACTS_DIRECTORY="$MOCK_DIRECTORY"/contracts
-CONTRACTS_COMPILED_DIRECTORY="$MOCK_DIRECTORY"/contracts_compiled
+CONTRACTS_DIRECTORY_V1="$MOCK_DIRECTORY"/contracts_v1
+CONTRACTS_DIRECTORY_V2="$MOCK_DIRECTORY"/contracts_v2
 
-# delete all artifacts except precompiled ones
-find $CONTRACTS_COMPILED_DIRECTORY -maxdepth 1 -type f -delete
+setup_asdf() {
+    if ! command -v asdf >/dev/null 2>&1; then
+        echo "Install asdf, before executing this script is required to manage scarb version"
+        exit 1
+    fi
+    asdf plugin-add scarb
+}
 
-# compile Cairo test contracts
-echo "Compiling Cairo contracts with $(poetry run starknet-compile-deprecated --version)"
+compile_contracts_with_scarb() {
+    CONTRACTS_DIRECTORY=$1
+    SCARB_WITH_VERSION=$(cat "$CONTRACTS_DIRECTORY/.tool-versions")
 
-number_of_contracts=0
-for contract in "$CONTRACTS_DIRECTORY"/*.cairo; do
-    basename=$(basename "$contract")
+    setup_asdf
```

Comment:
you add a plugin and install scarb each time the script is invoked, this has to be changed. Let's move this to another function (eg `ensure_scarb`?) that will check if required scarb version (and asdf plugin) is installed, and only installs it if it is not.

Reply:
I assume that asdf manages it. In case when plugin and scarb are instaled asdf returns 
```
Plugin named scarb already added
scarb 2.6.3 is already installed
```

But if it's not enough I will add `ensure_scarb`

Reply:
let's not depend on external binary (asdf) with that. check the installed versions (should be rather easy using asdf), and only act depending on the output